packages/cw-orch-fns-derive/src/query_fns.rs

```diff
@@ -157,5 +157,137 @@ pub fn query_fns_derive(input: ItemEnum) -> TokenStream {
         #derived_trait_impl
     );
 
-    expand.into()
+    expand
+}
+
+pub fn async_query_fns_derive(input: ItemEnum) -> proc_macro2::TokenStream {
```

Comment:
Could we have a common function with query_fns_derive, with only some differences (for maintainability) because this seems very close to the sync one